pkg/builder/common.go

```diff
@@ -52,6 +52,14 @@ func ApplyTemplate(u *v1alpha1.Build, tmpl v1alpha1.BuildTemplateInterface) (*v1
 		}
 	}
 
+	build = ApplyReplacements(build, replacements)
+	return build, nil
+}
+
+// ApplyReplacements replaces placeholders for declared parameters with the specified replacements.
+func ApplyReplacements(build *v1alpha1.Build, replacements map[string]string) *v1alpha1.Build {
```

Comment:
@dlorenc : Now that `ApplyReplacements ` is a public function , can you add test which shows the function expected behavior.

Reply:
Done! I left the old test cases, but I can look into removing/extracting them a bit more.

Reply:
Thank you :)